src/common/LatencyTuner.cpp

```diff
@@ -37,37 +37,39 @@ Result LatencyTuner::tune() {
         reset();
     }
 
-    switch (mState) {
-        case State::Idle:
-            if (--mIdleCountDown <= 0) {
-                mState = State::Active;
-            }
-            mPreviousXRuns = mStream.getXRunCount();
-            if (mPreviousXRuns < 0) {
-                result = static_cast<Result>(mPreviousXRuns); // error code
-                mState = State::Unsupported;
-            }
-            break;
+    // Set state to Active if the idle countdown has reached zero.
+    if (mState == State::Idle && --mIdleCountDown <= 0) {
+        mState = State::Active;
+    }
+
+    // When state is Active attempt to change the buffer size if the number of xRuns has increased.
+    if (mState == State::Active) {
```

Comment:
In the previous version the code was setting state to `Active` and then breaking out of `case`. In this edition the behavior changes. Is it intentional?

Reply:
Good spot. I've amended to keep in line with the old behaviour. 

With hindsight it might have been better to let @philburk refactor this code. It probably wants splitting into smaller methods.